src/dialogs/restart/dialog-restart.ts

```diff
@@ -0,0 +1,287 @@
+import { showToast } from "../../util/toast";
+import {
+  showAlertDialog,
+  showConfirmationDialog,
+} from "../generic/show-dialog-box";
+import { RestartDialogParams } from "./show-dialog-restart";
+
+@customElement("dialog-restart")
+class DialogRestart extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _params?: RestartDialogParams;
+
+  public async showDialog(params: RestartDialogParams): Promise<void> {
+    this._params = params;
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render(): TemplateResult {
+    if (!this._params) {
+      return html``;
+    }
+
+    return html`
+      <ha-dialog
+        open
+        @closed=${this.closeDialog}
+        hideActions
+        .heading=${createCloseHeading(
+          this.hass,
+          this.hass.localize("ui.dialogs.restart.heading")
+        )}
+      >
+        <mwc-list>
+          <ha-list-item
+            graphic="avatar"
+            twoline
+            hasMeta
+            @request-selected=${this._reload}
+          >
+            <div slot="graphic" class="icon-background reload">
+              <ha-svg-icon .path=${mdiAutoFix}></ha-svg-icon>
+            </div>
+            <span
+              >${this.hass.localize("ui.dialogs.restart.reload.title")}</span
+            >
+            <span slot="secondary"
+              >${this.hass.localize(
+                "ui.dialogs.restart.reload.short_description"
+              )}</span
+            >
+          </ha-list-item>
+          <ha-list-item
+            graphic="avatar"
+            twoline
+            hasMeta
+            @request-selected=${this._restart}
+          >
+            <div slot="graphic" class="icon-background restart">
+              <ha-svg-icon .path=${mdiRefresh}></ha-svg-icon>
+            </div>
+            <span
+              >${this.hass.localize("ui.dialogs.restart.restart.title")}</span
+            >
+            <span slot="secondary"
+              >${this.hass.localize(
+                "ui.dialogs.restart.restart.short_description"
+              )}</span
+            >
+          </ha-list-item>
+          <div class="divider"></div>
+          <ha-list-item
+            graphic="avatar"
+            twoline
+            hasMeta
+            @request-selected=${this._hostReboot}
+          >
+            <div slot="graphic" class="icon-background reboot">
+              <ha-svg-icon .path=${mdiPowerCycle}></ha-svg-icon>
+            </div>
+            <span
+              >${this.hass.localize("ui.dialogs.restart.reboot.title")}</span
+            >
+            <span slot="secondary"
+              >${this.hass.localize(
+                "ui.dialogs.restart.reboot.short_description"
+              )}</span
+            >
+          </ha-list-item>
+          <ha-list-item
```

Comment:
This entry requires the Supervisor

Reply:
I added the condition to display the reboot/shutdown and reload.